### PR TITLE
Log Run Modal Design Mismatch

### PR DESCRIPTION
**Pull request by sam-kakengi**

### CODE REVIEW

**Comment by sam-kakengi** [comment]
Screenshot 2024-10-10 164147

**Comment by Dimwiddle** [comment]
**General Feedback**:

- Can you change the asterisk to be red?
- Cross of the tickboxes in the Issue task to ensure requirements have been tested locally.

@sam-kakengi

**Comment by sam-kakengi** [comment]
Screenshot 2024-10-10 172824

How's this?

**Comment by sam-kakengi** [comment]
Oh crap, just adding the toast, hold on!

**Comment by Dimwiddle** [comment]
> Screenshot 2024-10-10 172824
> 
> 
> 
> How's this?

Looks better! It's a bit dark though, maybe a shade lighter would be good.

Happy for you to pick the shade 👍

File: frontend/src/components/core-components/dashboard/LogRun.js [hunk 1]
```diff
@@ -172,17 +138,24 @@ const LogRunModal = ({ open, handleClose }) => {
...
+                      required
+                      
                         
                     />
                   </Grid>
 
                   <Grid item xs={12} sm={6}>
                     <FormControl fullWidth size={isMobile ? 'small' : 'medium'}>
-                      <InputLabel id="route-label">Route</InputLabel>
+                      <InputLabel required id="route-label">Route</InputLabel>
```
Comment (Dimwiddle) [inline frontend/src/components/core-components/dashboard/LogRun.js:158]: I can't see on this file - has this `required` been added to all the required fields, with asterisk?

Does it highlight the field if it's not included when submitting the form.

**Comment by Dimwiddle** [comment]
> Screenshot 2024-10-10 174024
> 
> How's this for the toast? It appears at the bottom-center of the screen for 3 seconds and an animation to make it zoom out.

It does look good - however Is this part of the MUI toast component library? Ideally we want something consistent to be used across the app.

There will/should be a toast shown on flows such as:
- Login success
- Start Timer activated
- Add a logged run
- Logout success

**Comment by sam-kakengi** [comment]
> > Screenshot 2024-10-10 174024
> > How's this for the toast? It appears at the bottom-center of the screen for 3 seconds and an animation to make it zoom out.
> 
> It does look good and a temporary is fine - however Is this part of the MUI toast component library? Ideally we want something consistent to be used across the app.
> 
> There will/should be a toast shown on flows such as:
> 
> * Login success
> * Start Timer activated
> * Add a logged run
> * Logout success
> 
> The toast components we use should be the same across all of those.

Yeah it's a good library from react-toastify not MUI; but it allows us to use these consistently across the app by including it within the App component (the heart of the app) and it's very easily customisable too. Great functionality.

**Comment by sam-kakengi** [comment]
I can easily add these toasts to all of the necessary features!

File: frontend/src/components/core-components/dashboard/LogRun.js [hunk 2]
```diff
@@ -55,26 +53,47 @@ const LogRunModal = ({ open, handleClose }) => {
...
+      value = value.replace(/[^0-9:]/g, "")
+
+      if (value.length === 2 || value.length === 5) {
+        value += value.endsWith(":") ? "" : ":"
+      }
+
+      value = value.slice(0, 8)
+
+      setRunData({ ...runData, runTime: value })
+    }
```
Comment (Dimwiddle) [inline frontend/src/components/core-components/dashboard/LogRun.js:70]: Can this be included in the timeUtil.js for easier repeat use?

File: frontend/src/components/core-components/dashboard/LogRun.js [hunk 3]
```diff
@@ -55,26 +53,47 @@ const LogRunModal = ({ open, handleClose }) => {
...
+        start: startDateTime,
+        finished: finishDateTime,
+        duration: durationInSeconds,
         route: runData.route,
         notes: runData.notes,
       }
   
       
       const response = await postRunData(formattedData)
+      console.log(response)
```
Comment (Dimwiddle) [inline frontend/src/components/core-components/dashboard/LogRun.js:91]: Can remove this logging.

**Comment by sam-kakengi** [comment]
Alternative is from the MUI Alerts component, but it has to be added separately to each individual component and it doesn't function as well as I've used it for the login functionality and it doesn't come with advanced features like animation, it messes with padding/margin of other elements also.

**Comment by Dimwiddle** [comment]
> Alternative is from the MUI Alerts component, but it has to be added separately to each individual component and it doesn't function as well as I've used it for the login functionality and it doesn't come with advanced features like animation, it messes with padding/margin of other elements also.

OK that's good to know. In that case you can keep with this toast 👌

Also there's a couple tweaks we need to do; so the other toasts can be done as part of that User Acceptance Test phase (before the Production deployment).

**Comment by sam-kakengi** [comment]
> > Alternative is from the MUI Alerts component, but it has to be added separately to each individual component and it doesn't function as well as I've used it for the login functionality and it doesn't come with advanced features like animation, it messes with padding/margin of other elements also.
> 
> OK that's good to know. In that case you can keep with this toast 👌
> 
> Also there's a couple tweaks we need to do; so the other toasts can be done as part of that User Acceptance Test phase (before the Production deployment).

Sweet sounds good

**Review by Dimwiddle — CHANGES_REQUESTED** [review]
Just a couple of minor tweaks. Otherwise it's looking good!

File: frontend/src/components/core-components/dashboard/LogRun.js (hunk 1 above)
Comment (sam-kakengi) [reply frontend/src/components/core-components/dashboard/LogRun.js:158]: Yes it has, the required field adds the asterix's onto the input labels; it doesn't for the route and date input's but it does for the duration textfield. Adding functionality for those two fields now

Comment (sam-kakengi) [reply frontend/src/components/core-components/dashboard/LogRun.js:158]: So datepicker is very annoying with it's error handling, as it seems I can't generate a custom error for it and it's in-built error functionality doesn't seem to be activating when needed - so an alternative I'm thinking I can just put in the default value for the date to always be the date of the day it is; so then there's no way to put an empty date in the field? What do you think? The other fields work great.

**Comment by Dimwiddle** [comment]
Have you seen this flickering of the datepicker on your side? @sam-kakengi 

I get this on Brave (Chromium)

https://github.com/user-attachments/assets/d2f39245-a831-454a-bd0b-7fc028a0a9b0

**Comment by sam-kakengi** [comment]
All done, when you can, take a look at the final changes!

File: frontend/src/components/core-components/dashboard/DashboardComp.js [hunk 4]
```diff
@@ -22,7 +24,54 @@ const DashboardComponent = ({setIsAuthenticated}) => {
...
+            const processedData = processData(response.data)
+            setChartData(processedData)
+          } else {
+            setChartError('No data available')
+          }
+        } catch (err) {
+          setChartError('Failed to fetch data')
+        } finally {
+          setLoading(false)
+          console.log(chartData)
```
Comment (Dimwiddle) [inline frontend/src/components/core-components/dashboard/DashboardComp.js:67]: Remove logging here (and anywhere else you see it)

**Review by Dimwiddle — APPROVED** [review]

**Pull request merged**